parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileReader.java

```diff
@@ -517,7 +524,74 @@ public PageReadStore readNextRowGroup() throws IOException {
     return columnChunkPageReadStore;
   }
 
+  /**
+   * Reads the requested predicate columns dictionaries from the row group at the current file position.
+   * Not reading all the requested columns, only read predicate columns.
+   * eg. select columnA, columnB, columnC from table where columnC > 10
+   *     only reads columnC's dictionary
+   * @param requestedSchema the predicate columns to read
+   * @throws IOException if an error occurs while reading
+   * @return the DictionaryPageReadStore which can provide DictionaryPageReaders for each column.
+   */
+  public DictionaryPageReadStore getCurrentRowGroupDictionaries(MessageType requestedSchema) throws IOException {
```

Comment:
@julienledem Unfortunately, I don't think this API works very well for the uses cases we have right now.  The issue @zhenxiao  is seeing is that he wants to identify the row groups not filtered prior to processing them.  This interface requires that you do both in lock step.  The filter api has the same problem in that it basically needs to do up front row group filtering without loading any column data.  We would have to be able to advance the dictionary separately from the row group, which would make this interface rather clunky. 

Overall, I think the static interface was closer to what we currently need (though I can see how this approach can be used to filter while processing).  Maybe we just need to separate out the reading of the dictionary as a utility that gets used by this.